graylog2-web-interface/src/stores/inputs/InputStatesStore.js

```diff
@@ -28,13 +28,53 @@ const InputStatesStore = Reflux.createStore({
             }
             result[input.id][node] = input;
           });
-        })
+        });
         this.inputStates = result;
         this.trigger({inputStates: this.inputStates});
 
         return result;
       });
-  }
+  },
+
+  start(input) {
+    let url;
+
+    if (input.global) {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.ClusterInputStatesController.start(input.id).url);
+    } else {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.InputStatesController.start(input.id).url);
+    }
+
+    return fetch('PUT', url)
+      .then(
+        () => {
+          this.list();
+          UserNotification.success(`Input '${input.title}' started successfully`);
```

Comment:
It turns out, it is difficult to know when an input is started or stopped in the server side. Maybe I'm missing something, but it looks like Guava's EventBus does not provide any way of checking the status of the listeners, and we are using an async event bus, which makes everything a bit more complicated, as the execution of listeners may be delayed.

In the end I did something similar to what you suggested, checking the result of the input states list request. I don't like that we don't provide feedback on the start/stop actions when using the REST API, but I guess we'll have to live with that for now.